haystack/preview/components/audio/whisper_remote.py

```diff
@@ -30,52 +28,85 @@ class RemoteWhisperTranscriber:
+            try:
+                api_key = os.environ["OPENAI_API_KEY"]
+            except KeyError as e:
+                raise ValueError(
+                    "RemoteWhisperTranscriber expects an OpenAI API key. "
+                    "Set the OPENAI_API_KEY environment variable (recommended) or pass it explicitly."
+                ) from e
+
+        self.organization = organization
         self.model_name = model_name
-        self.api_key = api_key
-        self.api_base = api_base
-        self.whisper_params = whisper_params or {}
+        self.api_base_url = api_base_url
 
-    @component.output_types(documents=List[Document])
-    def run(self, audio_files: List[Path], whisper_params: Optional[Dict[str, Any]] = None):
-        """
-        Transcribe the audio files into a list of Documents, one for each input file.
+        # Only response_format = "json" is supported
+        whisper_params = kwargs
+        whisper_params["response_format"] = "json"
+        self.whisper_params = whisper_params
 
-        For the supported audio formats, languages, and other parameters, see the
-        [Whisper API documentation](https://platform.openai.com/docs/guides/speech-to-text) and the official Whisper
-        [github repo](https://github.com/openai/whisper).
+        openai.api_key = api_key
+        if organization is not None:
+            openai.organization = organization
 
-        :param audio_files: a list of paths or binary streams to transcribe
-        :returns: a list of Documents, one for each file. The content of the document is the transcription text,
-            while the document's metadata contains all the other values returned by the Whisper model, such as the
-            alignment data. Another key called `audio_file` contains the path to the audio file used for the
-            transcription.
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+        This method overrides the default serializer in order to
+        avoid leaking the `api_key` value passed to the constructor.
         """
-        if whisper_params is None:
-            whisper_params = self.whisper_params
+        return default_to_dict(
+            self,
+            model_name=self.model_name,
+            organization=self.organization,
+            api_base_url=self.api_base_url,
+            **self.whisper_params,
+        )
 
-        documents = self.transcribe(audio_files, **whisper_params)
-        return {"documents": documents}
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "RemoteWhisperTranscriber":
+        """
+        Deserialize this component from a dictionary.
+        """
+        return default_from_dict(cls, data)
 
-    def transcribe(self, audio_files: Sequence[Union[str, Path, BinaryIO]], **kwargs) -> List[Document]:
+    @component.output_types(documents=List[Document])
+    def run(self, audio_files: List[Union[str, Path, BinaryIO]]):
```

Comment:
If it can work, I'd propose to change the signature of run in this way:

```python
 def run(self,  streams: List[ByteStream]):
```

You can find `ByteStream` here: https://github.com/deepset-ai/haystack/blob/main/haystack/preview/dataclasses/byte_stream.py  It's a recent addition, we are trying to make all components that accept paths and stream use this abstraction going forward. Naturally, you may need to adapt the rest of the method for this to work.

Let me know if you encounter problems with it!

Reply:
Thanks! I will adapt this method to the new signature. 

In the previous implementation, we store the file path of the audio file in the Document metadata. In this case, since we are using the ByteStream object directly, that information is lost. 

Perhaps we can store the `file_name` and `file_path` as part of the `metadata` attribute of the ByteStream object. This way we can directly pass this `metadata` when creating the Document.

Reply:
Oh, you're right about losing the path information.

Storing it in the metadata would be a good patch for now, but let me first check with my team mates whether we should rather add an optional `path` field to `ByteStream`. I'll be back soon with an answer!

Reply:
https://github.com/deepset-ai/haystack/pull/6168/

Reply:
Ok, we decided to use the metadata for this usecase :slightly_smiling_face: